CustomerOrderDataBase.java

```diff
@@ -161,9 +161,162 @@ else if (input.contains("d")) {
+                for (int i = 0; i < cust_email.length(); i++) {
+                    if (cust_email.charAt(i) == '@') {
+                        countr++;
+                        if (countr > 1) {
+                            flag = false;
+                            break;
+                        }
+                        if (i >= 1) flag = true;
+                        else {
+                            flag = false;
+                            break;
+                        }
 
+                    }
+                    if (cust_email.charAt(i) == '.') {
+                        countd++;
+                        if (countd > 1) {
+                            flag = false;
+                            break;
+                        }
+                        if (i >= 3) flag = true;
+                        else {
+                            flag = false;
+                            break;
+                        }
+                    }
+                    if (cust_email.indexOf(".") - cust_email.indexOf("@") >= 2) {
+                        flag = true;
+                    }
+                    if (!flag) break;
+                }
+                if (flag && cust_email.length() >= 5) {
+                    break;
+                } else {
+                    System.out.println("Invalid Email");
+                }
+            }
+
+            //Validate the customer ZIP code
+            System.out.print("Enter ZIP Code: ");
+            cust_location = console.next();
+            while((cust_location.length()) != 5){
+                System.out.println("ZIP Code must be 8 characters long");
+                System.out.print("Enter ZIP Code: ");
+                cust_location = console.next();
+            }
+
+            // Validate product id
+            System.out.print("Enter Product ID: ");
+            product_id = console.next();
+            while ((product_id.length()) != 12) {
+                System.out.println("Product ID must be 12 characters long!");
+                System.out.print("Enter Product ID: ");
+                product_id = console.next();
+            }
+
+            // Validate quantity
+            System.out.print("Enter Quantity: ");
+            while (!console.hasNextInt()) {
+                System.out.println("Quantity must be a whole number!");
+                System.out.print("Enter Quantity: ");
+                console.next();
+            }
+            quantity = console.nextInt();
+
+            //Confirming Entries
+            System.out.println(s);
+            System.out.println("You entered the following values:");
+            System.out.println(s);
+            System.out.printf("%11s %15s %12s %18s %15s\n", "|DATE:|", "|CUSTOMER EMAIL:|", "|CUSTOMER LOCATION:|", "|PRODUCT ID:|", "|QUANTITY:|");
+            System.out.printf("%11s %15s %12s %18s %15s\n", date, cust_email, cust_location, product_id, quantity);
```

Comment:
```suggestion
            System.out.printf("%-11s %-20s %-20s  %-18s %-15s\n", "|DATE:|", "|CUSTOMER EMAIL:|", "|CUSTOMER LOCATION:|", "|PRODUCT ID:|", "|QUANTITY:|");
            System.out.printf("%-11s %-20s %-20s  %-18s %15s\n", date, cust_email, cust_location, product_id, quantity);
```

Reply:
Fixing column width so headers have enough room to line up
Left justify headers and inputs besides quantity